contrib/inventory/ec2.py

```diff
@@ -672,7 +672,9 @@ def get_elasticache_clusters_by_region(self, region):
                 error = self.get_auth_error_message()
             if not e.reason == "Forbidden":
                 error = "Looks like AWS ElastiCache is down:\n%s" % e.message
-            self.fail_with_error(error, 'getting ElastiCache clusters')
+                self.fail_with_error(error, 'getting ElastiCache clusters')
+            else:
+                return
```

Comment:
If I have a user with no RDS permissions, for example, and no RDS instances, I consider it a gross violation of least surprise that ec2.py errors out rather than working as expected.